asm2/main.js

```diff
@@ -0,0 +1,49 @@
+function openForm() {
+    document.getElementById("myForm").style.display = "block";
+  }
+  
+function closeForm(){
+    document.getElementById("myForm").style.display = "none";
+}
+
+function onFormSubmit() {
+    var formData = readFormData();
+    insertNewRecord(formData);    
+}
+function calTotal(){
+    var formData = readFormData();
+    calTotal(formData)
+}
+function readFormData() {
+    var formData = {};
+    formData["ProdName"] = document.getElementById("ProdName").value;
+    formData["price"] = document.getElementById("price").value;
+    formData["quantity"] = document.getElementById("quantity").value;
+    formData["date"] = document.getElementById("date").value;
+    return formData;
```

Comment:
The formData object called JSON, JSON organized data in pair key-value format.
Each key-value pair consists of a key (a string) and a value (which can be a string, number, boolean, null, array, another JSON object, ...)